src/librustc_trans/back/link.rs

```diff
@@ -676,8 +679,12 @@ fn link_natively(sess: &Session,
     } else {
         &sess.target.target.options.pre_link_objects_dll
     };
-    for obj in pre_link_objects {
-        cmd.arg(root.join(obj));
+    // When dynamically linking to MUSL we don't need to pass our startup
+    // objects as those will be provided by the MUSL toolchain
+    if !musl || crt_static {
```

Comment:
Hm this is always something I *wish* we could sink into the target specs rather than have here... Does anything leap to mind for you as an easy way to do that?